instrumentation/hikaricp/src/main/java/com/splunk/opentelemetry/hikaricp/HikariPoolInstrumentation.java

```diff
@@ -16,37 +16,39 @@
 
 package com.splunk.opentelemetry.hikaricp;
 
-import static net.bytebuddy.matcher.ElementMatchers.isConstructor;
 import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
 
-import com.zaxxer.hikari.HikariDataSource;
+import com.zaxxer.hikari.metrics.MetricsTrackerFactory;
 import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
 import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
 import net.bytebuddy.asm.Advice;
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.matcher.ElementMatcher;
-import org.slf4j.LoggerFactory;
 
-class HikariDataSourceInstrumentation implements TypeInstrumentation {
+class HikariPoolInstrumentation implements TypeInstrumentation {
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return named("com.zaxxer.hikari.HikariDataSource");
+    return named("com.zaxxer.hikari.pool.HikariPool");
   }
 
   @Override
   public void transform(TypeTransformer transformer) {
     transformer.applyAdviceToMethod(
-        isConstructor(), this.getClass().getName() + "$ConstructorAdvice");
+        named("setMetricsTrackerFactory")
```

Comment:
What if this method is never called? :)

Reply:
It's always called by in the (only) constructor of the class, I've added a comment that explains this choice.